core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -7683,7 +7720,7 @@ private void addIdentifierOrdinal2ExpandSet(@UnknownInitialization ExtendedExpan
         for (SqlNode node : rowOperandList) {
           addIdentifierOrdinal2ExpandSet(node);
         }
-      } else if (sqlNode.getKind() == SqlKind.IDENTIFIER || sqlNode.getKind() == SqlKind.LITERAL) {
+      } else if (sqlNode.getKind() == SqlKind.LITERAL) {
```

Comment:
IDENTIFIER was not useful, since the aliasOrdinal set should only contain ordinals.

Reply:
+1